edn/src/symbols.rs

```diff
@@ -98,12 +101,14 @@ impl Keyword {
 }
 
 impl NamespacedKeyword {
-    pub fn new(namespace: &str, name: &str) -> Self {
-        assert!(!name.is_empty(), "Keywords cannot be unnamed.");
-        assert!(!namespace.is_empty(), "Keywords cannot have an empty non-null namespace.");
+    pub fn new<T>(namespace: T, name: T) -> Self where T: Into<String> {
```

Comment:
I think it would be in general-purpose EDN; Clojure certainly accepts it. I don't think it's useful for our purposes.

This kind of question is why I mentioned perhaps layering reversal on top of the EDN crate — EDN doesn't have a notion of attributes, let alone reversal! — rather than doing what's convenient for us. If we continue to have reversal as part of the EDN crate, then we should reject `:foo/_`. If we layer it on top, then not all valid EDN keywords are valid input to Mentat.